src/boot/efi/boot.c

```diff
@@ -2222,6 +2270,125 @@ static void config_entry_add_linux(
+        if (!keys_dir)
+                return log_oom();
+
+        err = open_directory(root_dir, keys_dir, &dir);
+        if (EFI_ERROR(err))
+                return log_error_status_stall(err, L"Failed opening keys directory %s: %r", keys_dir, err);
+
+        /* we start by loading ALL the variables from the disk */
+        for (UINTN i = 0; i < ELEMENTSOF(sb_vars); i++) {
+                err = file_read(dir, sb_vars[i].filename, 0, 0, &sb_vars[i].buffer, &sb_vars[i].size);
+                if (EFI_ERROR(err)) {
+                        log_error_stall(L"Failed reading file %s: %r", sb_vars[i].filename, err);
+                        goto out_deallocate;
+                }
+        }
+
+        /* we then write them to the efivars */
+        for (UINTN i = 0; i < ELEMENTSOF(sb_vars); i++) {
+                err = efivar_set_raw(&sb_vars[i].vendor, sb_vars[i].name, sb_vars[i].buffer, sb_vars[i].size, sb_vars_opts);
+                if (EFI_ERROR(err)) {
+                        log_error_stall(L"Failed to write %s secure boot variable: %r", sb_vars[i].name, err);
+                        goto out_deallocate;
+                }
+        }
+
+        /* if we loaded the signature keys successfully then we remove all the
+         * signature keys entries from  the menu as the system is now locked down. */
+        config_remove_by_type(config, LOADER_SIGNATURE_KEYS);
+
+out_deallocate:
+        /* we deallocate the buffers storing the UEFI variables */
+        for (UINTN i = 0; i < ELEMENTSOF(sb_vars); i++) {
+                if (sb_vars[i].buffer)
+                        FreePool(sb_vars[i].buffer);
+                else
+                        break;
+        }
+
+        return err;
+}
+
+static ConfigEntry *config_entry_add_signature_keys(
+                Config *config,
+                const CHAR16 *name) {
+        assert(config);
+        assert(name);
+
+        ConfigEntry *entry = NULL;
+        _cleanup_freepool_ CHAR16 *title = NULL, *id = NULL;
+        CHAR16 *current_name = NULL;
+
+        id = PoolPrint(L"signature-keys-%s", name);
+        if (!id) {
+                log_oom();
+                return NULL;
+        }
+
+        title = PoolPrint(L"Enroll signature keys: %s", name);
+        if (!title) {
+                log_oom();
+                return NULL;
+        }
+
+        entry = AllocatePool(sizeof(ConfigEntry));
+        if (!entry) {
+                log_oom();
+                return NULL;
+        }
+
+        current_name = StrDuplicate(name);
+        if (!current_name) {
+                log_oom();
+                return NULL;
+        }
+
+        *entry = (ConfigEntry) {
+                .id = TAKE_PTR(id),
+                .title = TAKE_PTR(title),
+                .type = LOADER_SIGNATURE_KEYS,
+                .call = secure_boot_enroll,
+                .tries_done = UINTN_MAX,
+                .tries_left = UINTN_MAX,
+                .current_name = current_name,
```

Comment:
Why set this? This is only used for boot counting and imho makes little sense for key enrollment. Also, this is totally unused since tries is `UINTN_MAX` anyway.